.azure-pipelines/env/decide-what-to-publish.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -e
+
+PUBLISH_GITHUB_RELEASE=$RUN_PACKING
+PUBLISH_DOCKERHUB=false
+
+if [[ "$ISSTABLEBRANCH" == True && -n "$POSTSUBMIT" && "$NOSYNC" != true ]]; then
+    # main
+    if [[ "$ISMAIN" == True ]]; then
+        # Update the Dockerhub README
+        PUBLISH_DOCKERHUB=true
+    # Not main, and not -dev
+    elif [[ $STATE_ISDEV == false ]]; then
+        if [[ $STATE_VERSIONPATCH -eq 0 ]]; then
+            # A just-forked branch
+            PUBLISH_GITHUB_RELEASE=false
+        fi
+    fi
+fi
+
+# Only run Envoy release CI if explictly set
+if { [[ "$PRESUBMIT" != 'true' ]] && [[ "$POSTSUBMIT" != 'true' ]]; } || [[ $BUILD_REASON == "Schedule" ]]; then
```

Comment:
Not sure where are `PRESUBMIT`, `POSTSUBMIT`, and `NOSYNC` defined?

Reply:
they are defined as azp library vars

Reply:
Thanks, I assume `ISSTABLEBRANCH` and `ISMAIN` are also azure ado pipeline lib vars, right?

Reply:
no - they are defined _in_ the pipeline - they are _not_ case sensitive - see, eg:

```
$ git grep -i ISSTABLEBRANCH
```

Reply:
Ah, I see. The ado pipeline syntax is a bit subtle. Found the doc: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#access-variables-through-the-environment

Reply:
I don't need to update the PR then, and the PR is ready to review, thanks!